plugins/nativeWebsocketPlugin.ts

```diff
@@ -0,0 +1,36 @@
+import { Plugin } from '@nuxt/types';
+import Vue from 'vue';
+// eslint-disable-next-line @typescript-eslint/ban-ts-comment -- Need to build a Type package for the module
+// @ts-ignore
+import VueNativeSock from 'vue-native-websocket';
+
+declare module '@nuxt/types' {
+  interface Context {
+    $connect: (url: string) => void;
+    $disconnect: () => void;
+  }
+
+  interface NuxtAppOptions {
+    $connect: (url: string) => void;
+    $disconnect: () => void;
+  }
+}
+
+declare module 'vue/types/vue' {
+  interface Vue {
+    $connect: (url: string) => void;
+    $disconnect: () => void;
+  }
+}
+
+const nativeSocketPlugin: Plugin = ({ store }) => {
+  Vue.use(VueNativeSock, 'ws://localhost', {
```

Comment:
Any reason why localhost and not 127.0.0.1?

Reply:
Not really. It just needs something there to initialize, or it errors out, so I put in something generic.

Reply:
I think we should use 127.0.0.1 over any dns query, we shouldn't rely on any DNS query for this.

We don't know for sure if the docker container of Vue has a dns resolver, for instance.

Reply:
`localhost` doesn't go through a DNS resolver, it's handled by the OS' IP stack and works everywhere. Tons of stuff relies on it as the standard loopback address.

Reply:
@MrTimscampi And that's exactly the problem with it, Windows for instance does interpret it as IPv6 instead of IPv4. Some Linux distros do the same. I had problems resolving it in the past in Ubuntu after uninstalling NetworkManager.

127.0.0.1 is truly 100% cross-platform and standalone.

Cases where you can't rely on ``localhost`` are really scarce, but that doesn't mean that using 127.0.0.1 is a moot point. At the end of the day, although it's negligible, we're doing 3 steps (take domain -> resolve to ip -> connect) vs 2 (take IP -> connect)

Reply:
In this case, it won't even try to connect (Auto-connect is disabled, since we need authentication to connect).

I changed it, just in case, but it's really not necessary here.